src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -131,17 +133,32 @@ void CoreWorkerDirectTaskSubmitter::RequestNewWorkerIfNeeded(
                            << status.ToString();
           }
         }
-      }));
-  pending_lease_requests_.insert(scheduling_key);
+      });
+
+  if (status.ok()) {
+    pending_lease_requests_.insert(scheduling_key);
+  } else {
+    if (lease_client != local_lease_client_) {
```

Comment:
Do we ever hit this case? I thought the async call would always succeed and you'd just get a error callback.

Reply:
Yeah I think you're right.